src/test/java/org/hibernate/infra/asciidoctor/extensions/customnumbering/ExampleNumberingTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Hibernate Infra - Asciidoctor extensions
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.infra.asciidoctor.extensions.customnumbering;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.net.URISyntaxException;
+import java.nio.file.Paths;
+import java.util.regex.Pattern;
+
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import org.asciidoctor.Asciidoctor;
+import org.asciidoctor.OptionsBuilder;
+import org.asciidoctor.SafeMode;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+/**
+ * @author Marko Bekhta
+ */
+public class ExampleNumberingTest {
+
+	private static final Pattern EXAMPLE_TITLE_PATTERN = Pattern.compile( "Example [\\d]+\\.[\\d]+: .*" );
+
+	private static Asciidoctor DOCTOR;
+
+	@BeforeClass
+	public static void initAsciidoctor() {
+		DOCTOR = Asciidoctor.Factory.create();
+		DOCTOR.javaExtensionRegistry().treeprocessor( CustomNumberingProcessor.class );
+	}
+
+	@Test
+	public void simpleExampleNumberingTest() throws URISyntaxException, IOException {
+		File file = Paths.get( ExampleNumberingTest.class.getClassLoader().getResource( "test.asciidoc" ).toURI() ).toFile();
+		try ( FileReader reader = new FileReader( file ) ) {
+			StringWriter writer = new StringWriter();
+			DOCTOR.convert( reader, writer, OptionsBuilder.options().safe( SafeMode.SAFE ).asMap() );
+
+			Document convertedDoc = Jsoup.parse( writer.toString() );
+
+			Elements examples = convertedDoc.getElementsByClass( "exampleblock" );
+
+			examples.forEach( this::assertExampleNumbering );
+
+		}
+	}
+
+	private void assertExampleNumbering(Element example) {
```

Comment:
> As for the "table issue" it will not appear for this test

Understood. But it would be nice to be sure we don't have a regression on this front!

As mentioned in another comment, I opened an asciidoctor-ant PR, we'll see how it goes. Thanks for confirming than upgrading asciidoctorj fixed the issue.